### PR TITLE
Add compiler setting for __APPLE_USE_RFC_3542 for CNIODarwin

### PR DESCRIPTION
**Pull request by glbrntt**
Motivation:

In #1888 a C compiler setting was added to Package.swift for CNIODarwin.
This isn't reflected in the associated CNIODarwin podspec so the pod
fails to build.

Modifications:

Modify the build_podspecs.sh script to add the missing compiler setting
for CNIODarwin.

Result:

CNIODarwin pod builds.

### CODE REVIEW

File: scripts/build_podspecs.sh [hunk 1]
```diff
@@ -83,6 +83,11 @@ for target in "${targets[@]}"; do
     libraries="s.libraries = 'z'"
   fi
 
+  compiler_flags=""
+  if [[ "$target" == "CNIODarwin" ]]; then
+    compiler_flags="s.compiler_flags = '-D__APPLE_USE_RFC_3542=1'"
```
Comment (weissi) [inline scripts/build_podspecs.sh:88]: @glbrntt if we set this unconditionally, why don't we just `#define __APPLE_USE_RFC_3542 1` before the `include` of the relevant header (that makes use of this)?

Comment (glbrntt) [reply scripts/build_podspecs.sh:88]: Ah good point! I went for parity with the Package.swift but this is better, thanks for the suggestion.

Comment (weissi) [reply scripts/build_podspecs.sh:88]: Then we can probably also just take it out of `Package.swift`, no?

Comment (glbrntt) [reply scripts/build_podspecs.sh:88]: That's right, yes.

Unfortunately I can't get this to build this way -- perhaps there's a reason it was defined in Package.swift to begin with. Will have another look later.

**Comment by Lukasa** [comment]
@glbrntt Where are we with this?

**Comment by glbrntt** [comment]
> @glbrntt Where are we with this?

I couldn't get this working with weissi's suggestion. I suggest we review it as it is.

**Review by Lukasa — APPROVED** [review]

**Pull request merged**